### DISCUSSION TITLE
Questions regarding best practices using both the MASS media player entities and the original

### DISCUSSION THREAD

**Discussion by FrankJaco** [music-assistant/support#882] [Q&A]
I am aware of the option to "hide" the original media players and use only the MASS created ones when first configuring MASS. I didn't choose that option as I have a fairly complex configuration using several ChromecastiAudio devices and several speaker groups along with a Plex server. My ultimate goal is to eliminate plex and do all my music needs with MASS.  But I am not ready to turn off plex just yet. 

In trying to build a similar structure to what I have set up in plex, I am using the custom "Mushroom media player" card as well as the custom "mini media player" card along with the "Grid" card and the "Conditional" card to make thing visible and invisible as desired. (e.g. to show only the active volume controls, other controls and full album covers, etc. etc.). It is the conditional states that have me stressed at the moment.... It seems that I have to manage the states of the original media players as well as the MASS-created ones independently..  Turning on the MASS one doesn't turn on the matching original one. But to get sound I need to turn on both the original and MASS players. An additional complication is that the same exact "Friendly name" is used for both the MASS player and the original this gives HA's visual editor fits in some places and it chooses the wrong entity. Yaml editing is the only way to ensure the right entity is place.

Has anyone had similar challenges?  Was it a mistake NOT to hide the original medial players on initial setup?  I got to believe hiding them would completely break my complex plex setup?  Is that true? 

Any tips or wisdom would be greatly appreciated!

Thx,
fj

**Comment by OzGav** [comment]
I am not sure it is true that the old and new players have to be on for sound to be heard. Have a look at what @marcelveldt said here https://github.com/music-assistant/hass-music-assistant/discussions/876 You might have to provide some more info about what players you are using.

**Comment by FrankJaco** [comment]
Thanks for the reply. I actually have made a lot of progress since I 
first wrote this.  I have basically 3 main ChromeCast-Audio devices 
driving various amps receivers, and speakers (Living Room, Hi Fi and 
Patio. I have and additional 3 main audio groups of these 
Chromcast-Audio devices (Inside House, Terrace Level, All Speakers).

What I did to make things work and manageable:

As mentioned having the original Chromecast devices named with the same 
friendly name as the MASS devices gave the visual editor issues. So I 
changed all of the MASS devices names to a similar but different name 
and MASS automatically changed its friendly name accordingly.  For 
example my MASS equivalent for ChromCast "Living Room" is now 
"LivingRoom" - I simply removed a space. This subtle change made all of 
the visual editor issues (in the grid, conditional and vertical stack 
cards) go away. I am not sure if Marcel has a reason that he uses the 
exact friendly name for MASS devices, if not, I think it would be a good 
change for him to make.  Something subtly different. Having the visual 
editor working more predictably coupled with a better understanding of 
"on" "off" "idle" etc. of both native Chromecasts and MASS equivalents 
have made creating the right logic in the conditional cards doable, 
although it is still more complex than chromecasts alone.

The need to turn on both the Chromecast devices and Mass devices so far 
seem to be limited to Pandora and a local cast-able media player (Pulsar 
on Android).  With pandora if Chromecast-Audio is turned off and Mass is 
in the idle state, the app on the phone will crash 75% of the time when 
attempting to cast.  The Chomecast audio device will chime indicating it 
is turning on, but nothing plays as the App has crashed. The other 25% 
of the time the App doesnt crash but no sound occurs and the progress 
bar on Pandora doesn't advance (So not crashed, locked up).

I have a script that I have been using for years that sets every media 
player to the Off state. I generally use this via a button as my master 
audio kill switch. This is to prevent a reset of apps to cause the 
chromecast to chime in the middle of the night. (My wife thanks me for 
this! :-)  I have now since modified this script to turn all media 
players off, delay 3 seconds then set only the 3 MASS devices to the 
"idle" state leaving everthing else off. This change seems to helped.  I 
also make a quick and easy tool to toggle the state of the actual 
Chromecast devices to the idle state. I now toggle the Chromecasts to 
"idle" just prior to playing from Pandora (and the Pulsar player). As 
long as I have done that so that both Chromecast and MASS devices are 
set to idle), I no longer have playback issues with either app.

Hopefully this long explanation is helpful to anyone struggling. Keep at 
it. It really is a great project and addition to HA!

cheers,

Frank

**Comment by OzGav** [comment]
Relates to v1
